homework/src/index.js

```diff
@@ -30,17 +32,109 @@
+    try {
+      const data = await fetchJSON(url);
+        data.sort((a, b) => a.name.localeCompare(b.name, 'fr', {ignorePunctuation: true}));
+        data.forEach((repository, index) => {
+          createAndAppend('option', select, {html: repository.name, value: index});
+        });
+        const container = createAndAppend('div', root, { id: 'container'});
+        contentLeft(container, data, select);
+        contentRight(container, data, select);
+    }
+    catch(error) {
+      createAndAppend('div', root, { html: error.message, class: 'alert-error' });
+    }
+  }
+  
+  // function who creates a left content section
+  function contentLeft(container, data, select) {
+    const leftSection = createAndAppend('div', container, { id: 'leftSection', class: 'whiteframe' });
+    showTableRepo(leftSection, data[0]);
+
+    select.addEventListener('change', (event) => {
+      leftSection.innerHTML = "";
+      const theEventIndex = event.target.value;
+      showTableRepo(leftSection, data[theEventIndex]);
+    });
+  }
+
+  // drow the table
+function showTableRepo(leftSection, repository) {
+  // create table of the contents
+  const lTable = createAndAppend('table', leftSection);
+  // line one
+  const tr1 = createAndAppend('tr', lTable);
+  createAndAppend('td', tr1, {html: 'Repository :', class: 'lable'});
+  const td1 = createAndAppend('td', tr1);
+  const nameLink = '<a href="' + repository.html_url + '" target="_blank">' + repository.name + '</a>';
+  createAndAppend('p', td1, {html: nameLink });
+  // line two
+  const tr2 = createAndAppend('tr', lTable);
+  createAndAppend('td', tr2, {html: 'Description :', class: 'lable'});
+  const td2 = createAndAppend('td', tr2);
+  createAndAppend('p', td2, {html: repository.description});
+  // line three
+  const tr3 = createAndAppend('tr', lTable);
+  createAndAppend('td', tr3, {html: 'Forks :', class: 'lable'});
+  const td3 = createAndAppend('td', tr3);
+  createAndAppend('p', td3, {html: repository.forks});
+  // line four
+  const tr4 = createAndAppend('tr', lTable);
+  createAndAppend('td', tr4, {html: 'Updated :', class: 'lable'});
+  const td4 = createAndAppend('td', tr4);
+  const defaultElDate = repository.updated_at;
+  const defaultDate = new Date(defaultElDate);
+  const endDate = defaultDate.toLocaleString();
+  createAndAppend('p', td4, {html: endDate});
+}
+
+  // function who creates a right content section
+  async function contentRight(container, data, select) {
+    const rightSection = createAndAppend('div', container, { id: 'rightSection', class: 'whiteframe' });
+    createAndAppend('h5', rightSection, {html: "Contributions :", class: 'contributionsLable'});
+    const contributorsURL = data[0].contributors_url;
+    try {
+      const contributors = await fetchJSON(contributorsURL);
+      showContributors(rightSection, contributors);
+    }
+    catch(error) {
+      createAndAppend('div', rightSection, { html: error.message, class: 'alert-error' });
```

Comment:
I think you have to return here before you got two different kind of errors, because the interpreter will continue running till the end of the function, and when ever we have an error we need to know exactly what is it.
so if an error happen somehow the `try catch` will prevent the program crashing and with that it will just continue till the end.